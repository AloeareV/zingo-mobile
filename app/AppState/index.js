import Address from './Address';
import AddressBookEntry from './AddressBookEntry';
import AppStateLoaded from './AppStateLoaded';
import AppStateLoading from './AppStateLoading';
import ConnectedCompanionApp from './ConnectedCompanionApp';
import ErrorModalData from './ErrorModalData';
import InfoType from './InfoType';
import PasswordState from './PasswordState';
import ReceivePageState from './ReceivePageState';
import SendJsonToType from './SendJsonToType';
import SendPageState from './SendPageState';
import SendProgress from './SendProgress';
import ServerSelectState from './ServerSelectState';
import SettingsFileEntry from './SettingsFileEntry';
import SyncStatus from './SyncStatus';
import SyncStatusReport from './SyncStatusReport';
import ToAddr from './ToAddr';
import TotalBalance from './TotalBalance';
import Transaction from './Transaction';
import TxDetailType from './TxDetailType';
import WalletSeed from './WalletSeed';
import WalletSettings from './WalletSettings';

export {
  AppStateLoaded,
  AppStateLoading,
  Address,
  AddressBookEntry,
  ConnectedCompanionApp,
  ErrorModalData,
  InfoType,
  PasswordState,
  ReceivePageState,
  SendJsonToType,
  SendPageState,
  SendProgress,
  ServerSelectState,
  SettingsFileEntry,
  SyncStatus,
  SyncStatusReport,
  ToAddr,
  TotalBalance,
  Transaction,
  TxDetailType,
  WalletSeed,
  WalletSettings,
};
