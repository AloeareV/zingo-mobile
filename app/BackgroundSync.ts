import RPCModule from './RPCModule';
import AsyncStorage from '@react-native-async-storage/async-storage';
import NetInfo, { NetInfoStateType } from '@react-native-community/netinfo';
import { RPCSyncStatusType } from './rpc/types/RPCSyncStatusType';

const BackgroundSync = async (task_data: any) => {
  // this not interact with the server.
  const exists = await RPCModule.walletExists();

  // only if exists the wallet file make sense to do the sync.
  if (exists && exists !== 'false') {
    // only if we have connection make sense to call RPCModule.
    const networkState = await NetInfo.fetch();
    if (
      !networkState.isConnected ||
      networkState.type === NetInfoStateType.cellular ||
      (networkState.details !== null && networkState.details.isConnectionExpensive)
    ) {
      //console.log(
      //  'BS: Not started (connected: ' + networkState.isConnected,
      //  +', type: ' +
      //    networkState.type +
      //    +', expensive connection: ' +
      //    networkState.details?.isConnectionExpensive +
      //    ')',
      //);
      return;
    }
    // if the App goes to Foreground kill the interval
    const background = await AsyncStorage.getItem('@background');
    if (background === 'no') {
      //console.log('BS: Not started (going to foreground)');
      return;
    }

    let batch_num = -1;
    console.log('BS:', task_data);

    // finishEarly has two fields: wait, and done.
    // wait() returns a promise, which is resolved when
    // done() is called
    let finishEarly = manuallyResolve();

    let saver = setInterval(async () => {
      const networkStateSaver = await NetInfo.fetch();
      if (
        !networkStateSaver.isConnected ||
        networkStateSaver.type === NetInfoStateType.cellular ||
        (networkStateSaver.details !== null && networkStateSaver.details.isConnectionExpensive)
      ) {
        //console.log(
        //  'BS: Interrupted (connected: ' + networkStateSaver.isConnected,
        //  +', type: ' +
        //    networkStateSaver.type +
        //    +', expensive connection: ' +
        //    networkStateSaver.details?.isConnectionExpensive +
        //    ')',
        //);
        clearInterval(saver);
        finishEarly.done();
        return;
      }
      // if the App goes to Foreground kill the interval
      const backgroundSaver = await AsyncStorage.getItem('@background');
      if (backgroundSaver === 'no') {
        clearInterval(saver);
        //console.log('BS: Finished (going to foreground)');
        finishEarly.done();
        return;
      }

      const syncStatusStr: string = await RPCModule.execute('syncstatus', '');
      if (syncStatusStr) {
        if (syncStatusStr.toLowerCase().startsWith('error')) {
          //console.log(`BS: Error sync status ${syncStatusStr}`);
          return;
        }
      } else {
        //console.log('BS: Internal Error sync status');
        return;
      }
      // TODO: verify this JSON parse
      const ss: RPCSyncStatusType = await JSON.parse(syncStatusStr);

      //console.log('BS:', ss);
      if (ss.batch_num && ss.batch_num > -1 && batch_num !== ss.batch_num) {
        await RPCModule.doSave();
        //console.log('BS: saving...');
        // update batch_num with the new value, otherwise never change
        batch_num = ss.batch_num;
      }
    }, 5000);

    await Promise.race([RPCModule.execute('sync', ''), finishEarly.wait()]);
    clearInterval(saver);
  } else {
    console.log('BS: wallet file does not exist');
  }
  //console.log('BS: Finished (end of syncing)');
};

export default BackgroundSync;

function manuallyResolve() {
  let resolve: Function;
  // new Promise takes a function as an argument. When that function is called
  // the promise resolves with the value output by that function.
  // By passing the function out of the promise, we can call it later
  // in order to resolve the promise at will
  const promise = new Promise(fun => {
    resolve = fun;
  });

  function done() {
    resolve();
  }

  function wait() {
    return promise;
  }

  return { wait, done };
}
