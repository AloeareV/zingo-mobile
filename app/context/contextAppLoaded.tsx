import React, { ReactNode } from 'react';
import { StackScreenProps } from '@react-navigation/stack';

import {
  AppStateLoaded,
  SyncingStatusReportClass,
  TotalBalanceClass,
  ReceivePageStateClass,
  ErrorModalDataClass,
  SendProgressClass,
  WalletSettingsClass,
  TransactionType,
  InfoType,
  SyncingStatusType,
  WalletSeedType,
  DimensionsType,
  AddressClass,
  AddressBookClass,
  zecPriceType,
  BackgroundType,
  SendPageStateClass,
  ToAddrClass,
  NetInfoType,
  BackgroundErrorType,
} from '../AppState';

export const defaultAppStateLoaded: AppStateLoaded = {
  navigation: {} as StackScreenProps<any>['navigation'],
  route: {} as StackScreenProps<any>['route'],
  dimensions: {} as DimensionsType,
  appState: '' as string,
  netInfo: {} as NetInfoType,

  syncingStatusReport: new SyncingStatusReportClass(),
  totalBalance: new TotalBalanceClass(),
  addressPrivateKeys: new Map(),
  addresses: [] as AddressClass[],
  addressBook: [] as AddressBookClass[],
  transactions: [] as TransactionType[],
  sendPageState: new SendPageStateClass(new ToAddrClass(0)),
  receivePageState: new ReceivePageStateClass(''),
  info: {} as InfoType,
  walletSettings: new WalletSettingsClass(),
  syncingStatus: {} as SyncingStatusType,
  errorModalData: new ErrorModalDataClass('', ''),
  sendProgress: new SendProgressClass(0, 0, 0),
  walletSeed: {} as WalletSeedType,
  isMenuDrawerOpen: false,
  selectedMenuDrawerItem: '' as string,
  aboutModalVisible: false,
  computingModalVisible: false,
  settingsModalVisible: false,
  infoModalVisible: false,
  rescanModalVisible: false,
  seedViewModalVisible: false,
  seedChangeModalVisible: false,
  seedBackupModalVisible: false,
  seedServerModalVisible: false,
  syncReportModalVisible: false,
  poolsModalVisible: false,
  newServer: '' as string,
  uaAddress: '' as string,

  server: '' as string,
  currency: '',
  language: 'en',

  zecPrice: {
    zecPrice: 0,
    date: 0,
  } as zecPriceType,
  sendAll: false,
  background: {
    batches: 0,
    date: 0,
  } as BackgroundType,

  translate: () => '',
  backgroundError: {} as BackgroundErrorType,
};

export const ContextAppLoaded = React.createContext(defaultAppStateLoaded);

type ContextProviderProps = {
  children: ReactNode;
  value: AppStateLoaded;
};

export const ContextAppLoadedProvider = ({ children, value }: ContextProviderProps) => {
  return <ContextAppLoaded.Provider value={value}>{children}</ContextAppLoaded.Provider>;
};
