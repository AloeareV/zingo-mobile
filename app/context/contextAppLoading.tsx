import React, { ReactNode } from 'react';
import { StackScreenProps } from '@react-navigation/stack';

import {
  AppStateLoading,
  DimensionsType,
  InfoType,
  TotalBalanceClass,
  WalletSeedType,
  zecPriceType,
  backgroundType,
} from '../AppState';

export const defaultAppStateLoading: AppStateLoading = {
  navigation: {} as StackScreenProps<any>['navigation'],
  route: {} as StackScreenProps<any>['route'],
  dimensions: {} as DimensionsType,
  appState: '' as string,

  screen: 0,
  actionButtonsDisabled: false,
  walletExists: false,
  walletSeed: {} as WalletSeedType,
  totalBalance: new TotalBalanceClass(),
  info: {} as InfoType,

  server: '' as string,
  currency: '',
  language: 'en',

  zecPrice: {
    zecPrice: 0,
    date: 0,
  } as zecPriceType,
  sendAll: false,
  background: {
    batches: 0,
    date: 0,
  } as backgroundType,

  translate: () => '',
};

export const ContextAppLoading = React.createContext(defaultAppStateLoading);

type ContextProviderProps = {
  children: ReactNode;
  value: AppStateLoading;
};

export const ContextAppLoadingProvider = ({ children, value }: ContextProviderProps) => {
  return <ContextAppLoading.Provider value={value}>{children}</ContextAppLoading.Provider>;
};
