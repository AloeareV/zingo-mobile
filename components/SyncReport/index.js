import SyncReport from './SyncReport';

export default SyncReport;
