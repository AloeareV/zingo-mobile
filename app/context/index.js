import { defaultAppStateLoaded, ContextAppLoaded, ContextAppLoadedProvider } from './contextAppLoaded';
import { defaultAppStateLoading, ContextAppLoading, ContextAppLoadingProvider } from './contextAppLoading';

export {
  defaultAppStateLoaded,
  ContextAppLoaded,
  ContextAppLoadedProvider,
  defaultAppStateLoading,
  ContextAppLoading,
  ContextAppLoadingProvider,
};
