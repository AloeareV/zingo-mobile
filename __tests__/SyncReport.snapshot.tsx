/**
 * @format
 */

import 'react-native';
import React from 'react';

import { create } from 'react-test-renderer';
import SyncReport from '../components/SyncReport';
import { ContextLoadedProvider } from '../app/context';

import {
  ErrorModalData,
  ReceivePageState,
  SendPageState,
  SendProgress,
  SyncStatusReport,
  ToAddr,
  TotalBalance,
  WalletSettings,
} from '../app/AppState';

jest.mock('@fortawesome/react-native-fontawesome', () => ({
  FontAwesomeIcon: '',
}));
jest.mock('react-native-localize', () => ({
  getNumberFormatSettings: () => {
    return {
      decimalSeparator: '.', // us
      groupingSeparator: ',', // us
    };
  },
}));
jest.useFakeTimers();

// test suite
describe('Component Info - test', () => {
  //snapshot test
  const state = {
    navigation: null,
    route: null,
    dimensions: {} as {
      width: number;
      height: number;
      orientation: 'portrait' | 'landscape';
      deviceType: 'tablet' | 'phone';
      scale: number;
    },

    syncStatusReport: new SyncStatusReport(),
    addressPrivateKeys: new Map(),
    addresses: [],
    addressBook: [],
    transactions: null,
    sendPageState: new SendPageState(new ToAddr(0)),
    receivePageState: new ReceivePageState(),
    rescanning: false,
    wallet_settings: new WalletSettings(),
    syncingStatus: null,
    errorModalData: new ErrorModalData(),
    txBuildProgress: new SendProgress(),
    walletSeed: null,
    isMenuDrawerOpen: false,
    selectedMenuDrawerItem: '',
    aboutModalVisible: false,
    computingModalVisible: false,
    settingsModalVisible: false,
    infoModalVisible: false,
    rescanModalVisible: false,
    seedViewModalVisible: false,
    seedChangeModalVisible: false,
    seedBackupModalVisible: false,
    seedServerModalVisible: false,
    syncReportModalVisible: false,
    poolsModalVisible: false,
    newServer: null,
    uaAddress: null,
    info: null,
    translate: () => 'translated text',
    totalBalance: new TotalBalance(),
  };
  test('Matches the snapshot Info', () => {
    const info: any = create(
      <ContextLoadedProvider value={state}>
        <SyncReport closeModal={() => {}} />
      </ContextLoadedProvider>,
    );
    expect(info.toJSON()).toMatchSnapshot();
  });
});
