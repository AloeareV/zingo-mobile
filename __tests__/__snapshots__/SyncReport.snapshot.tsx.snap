// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component SyncReport - test Matches the snapshot SyncReport 1`] = `
<RCTSafeAreaView
  style={
    {
      "alignItems": "stretch",
      "backgroundColor": "rgb(242, 242, 242)",
      "display": "flex",
      "height": "100%",
      "justifyContent": "flex-start",
    }
  }
>
  <View
    style={
      {
        "alignItems": "center",
        "backgroundColor": "rgb(255, 255, 255)",
        "display": "flex",
        "paddingBottom": 0,
        "paddingTop": 10,
        "zIndex": -1,
      }
    }
    testID="header"
  >
    <Image
      source={
        {
          "testUri": "../../../assets/img/logobig-zingo.png",
        }
      }
      style={
        {
          "height": 80,
          "resizeMode": "contain",
          "width": 80,
        }
      }
    />
    <View
      style={
        {
          "height": 20,
        }
      }
    />
    <View
      style={
        {
          "alignItems": "center",
          "display": "flex",
          "flexDirection": "row",
          "flexWrap": "wrap",
          "justifyContent": "center",
          "marginVertical": 5,
        }
      }
    >
      <View
        style={
          {
            "alignItems": "center",
            "display": "flex",
            "flexDirection": "column",
            "flexWrap": "wrap",
            "justifyContent": "center",
          }
        }
      >
        <Text
          style={
            {
              "color": "rgb(28, 28, 30)",
              "fontSize": 18,
              "fontWeight": "600",
              "opacity": 1,
              "paddingHorizontal": 5,
            }
          }
        >
          translated text
        </Text>
      </View>
    </View>
    <View
      style={
        {
          "alignItems": "flex-end",
          "padding": 15,
          "position": "absolute",
          "right": 0,
        }
      }
    >
      <Text
        style={
          {
            "color": "rgb(216, 216, 216)",
            "fontSize": 8,
          }
        }
      >
        translated text
      </Text>
      <Text
        style={
          {
            "color": "rgb(216, 216, 216)",
            "fontSize": 8,
          }
        }
      >
        (undefinedxundefined)-undefined
      </Text>
    </View>
    <View
      style={
        {
          "backgroundColor": "rgb(0, 122, 255)",
          "height": 1,
          "width": "100%",
        }
      }
    />
  </View>
  <RCTScrollView
    contentContainerStyle={
      {
        "alignItems": "stretch",
        "flexDirection": "column",
        "justifyContent": "flex-start",
      }
    }
    style={
      {
        "maxHeight": "85%",
      }
    }
  >
    <View>
      <View
        style={
          {
            "alignItems": "flex-end",
            "display": "flex",
            "flexDirection": "row",
            "marginHorizontal": 20,
          }
        }
      >
        <View
          style={
            {
              "display": "flex",
              "marginTop": 20,
            }
          }
        >
          <Text
            style={
              {
                "color": "rgb(28, 28, 30)",
                "opacity": 0.65,
              }
            }
          >
            translated text
          </Text>
          <View
            style={
              {
                "width": 10,
              }
            }
          />
          <View
            style={
              {
                "display": "flex",
                "flexDirection": "row",
              }
            }
          >
            <Text
              style={
                {
                  "color": "red",
                  "fontSize": 18,
                  "fontWeight": "600",
                  "opacity": 1,
                }
              }
            >
               
              translated text
               
            </Text>
          </View>
        </View>
        <
          color="red"
          icon={
            {
              "icon": [
                640,
                512,
                [
                  62337,
                  "cloud-download",
                  "cloud-download-alt",
                ],
                "f0ed",
                "M144 480C64.5 480 0 415.5 0 336c0-62.8 40.2-116.2 96.2-135.9c-.1-2.7-.2-5.4-.2-8.1c0-88.4 71.6-160 160-160c59.3 0 111 32.2 138.7 80.2C409.9 102 428.3 96 448 96c53 0 96 43 96 96c0 12.2-2.3 23.8-6.4 34.6C596 238.4 640 290.1 640 352c0 70.7-57.3 128-128 128H144zm79-167l80 80c9.4 9.4 24.6 9.4 33.9 0l80-80c9.4-9.4 9.4-24.6 0-33.9s-24.6-9.4-33.9 0l-39 39V184c0-13.3-10.7-24-24-24s-24 10.7-24 24V318.1l-39-39c-9.4-9.4-24.6-9.4-33.9 0s-9.4 24.6 0 33.9z",
              ],
              "iconName": "cloud-arrow-down",
              "prefix": "fas",
            }
          }
          size={20}
          style={
            {
              "marginBottom": 5,
              "marginLeft": 5,
            }
          }
        />
      </View>
      <View
        style={
          {
            "display": "flex",
            "margin": 20,
            "marginBottom": 30,
          }
        }
      >
        <View
          style={
            {
              "display": "flex",
              "marginTop": 20,
            }
          }
        >
          <Text
            style={
              {
                "color": "rgb(28, 28, 30)",
                "opacity": 0.65,
              }
            }
          >
            Sync ID
          </Text>
          <View
            style={
              {
                "width": 10,
              }
            }
          />
          <Text
            style={
              {
                "color": "rgb(28, 28, 30)",
                "fontSize": 18,
                "fontWeight": "600",
                "opacity": 1,
              }
            }
          >
            1 - (translated text)
          </Text>
        </View>
        <View
          style={
            {
              "backgroundColor": "white",
              "height": 2,
              "marginBottom": 10,
              "marginTop": 15,
              "width": "100%",
            }
          }
        />
        <View
          style={
            {
              "display": "flex",
              "flexDirection": "row",
              "justifyContent": "space-between",
              "marginTop": 10,
              "width": "100%",
            }
          }
        >
          <Text
            style={
              {
                "color": "rgb(0, 122, 255)",
              }
            }
          >
            0
          </Text>
          <Text
            style={
              {
                "color": "rgb(0, 122, 255)",
              }
            }
          >
            500K
          </Text>
          <Text
            style={
              {
                "color": "rgb(0, 122, 255)",
              }
            }
          >
            1M
          </Text>
          <Text
            style={
              {
                "color": "rgb(0, 122, 255)",
              }
            }
          >
            1.5M
          </Text>
          <Text
            style={
              {
                "color": "rgb(0, 122, 255)",
              }
            }
          >
            2M
          </Text>
        </View>
        <View
          style={
            {
              "display": "flex",
              "flexDirection": "row",
              "width": "100%",
            }
          }
        >
          <View
            style={
              {
                "borderLeftColor": "rgb(0, 122, 255)",
                "borderLeftWidth": 1,
                "borderRightColor": "rgb(0, 122, 255)",
                "borderRightWidth": 1,
                "height": 10,
                "width": "25%",
              }
            }
          />
          <View
            style={
              {
                "borderLeftColor": "rgb(0, 122, 255)",
                "borderLeftWidth": 1,
                "borderRightColor": "rgb(0, 122, 255)",
                "borderRightWidth": 1,
                "height": 10,
                "width": "25%",
              }
            }
          />
          <View
            style={
              {
                "borderLeftColor": "rgb(0, 122, 255)",
                "borderLeftWidth": 1,
                "borderRightColor": "rgb(0, 122, 255)",
                "borderRightWidth": 1,
                "height": 10,
                "width": "25%",
              }
            }
          />
          <View
            style={
              {
                "borderLeftColor": "rgb(0, 122, 255)",
                "borderLeftWidth": 1,
                "borderRightColor": "rgb(0, 122, 255)",
                "borderRightWidth": 1,
                "height": 10,
                "width": "25%",
              }
            }
          />
        </View>
        <View
          style={
            {
              "borderBottomColor": "rgb(0, 122, 255)",
              "borderBottomWidth": 2,
              "display": "flex",
              "flexDirection": "row",
              "justifyContent": "flex-start",
              "marginBottom": 0,
              "width": "100%",
            }
          }
        >
          <View
            style={
              {
                "backgroundColor": "blue",
                "borderLeftColor": "rgb(0, 122, 255)",
                "borderLeftWidth": 1,
                "borderRightColor": "blue",
                "borderRightWidth": 1,
                "height": 10,
                "width": "75.00505%",
              }
            }
          />
          <View
            style={
              {
                "backgroundColor": "yellow",
                "borderBottomColor": "blue",
                "borderBottomWidth": 5,
                "borderRightColor": "yellow",
                "borderRightWidth": 1,
                "height": 10,
                "width": "19.99995%",
              }
            }
          />
          <View
            style={
              {
                "backgroundColor": "transparent",
                "borderRightColor": "rgb(0, 122, 255)",
                "borderRightWidth": 1,
                "height": 10,
                "width": "4.995%",
              }
            }
          />
        </View>
        <View
          style={
            {
              "alignItems": "center",
              "display": "flex",
              "flexDirection": "row",
              "justifyContent": "flex-start",
              "marginTop": 5,
              "width": "100%",
            }
          }
        >
          <Text
            style={
              {
                "color": "rgb(0, 122, 255)",
              }
            }
          >
            translated text
          </Text>
          <View
            style={
              {
                "backgroundColor": "blue",
                "display": "flex",
                "flexDirection": "row",
                "height": 10,
                "justifyContent": "flex-start",
                "margin": 5,
                "width": 10,
              }
            }
          />
          <Text
            style={
              {
                "color": "rgb(28, 28, 30)",
              }
            }
          >
            1900100translated text
          </Text>
        </View>
        <View
          style={
            {
              "alignItems": "center",
              "display": "flex",
              "flexDirection": "row",
              "justifyContent": "flex-start",
              "marginTop": 5,
              "width": "100%",
            }
          }
        >
          <Text
            style={
              {
                "color": "rgb(0, 122, 255)",
              }
            }
          >
            translated text
          </Text>
          <View
            style={
              {
                "backgroundColor": "yellow",
                "display": "flex",
                "flexDirection": "row",
                "height": 10,
                "justifyContent": "flex-start",
                "margin": 5,
                "width": 10,
              }
            }
          />
          <Text
            style={
              {
                "color": "rgb(28, 28, 30)",
              }
            }
          >
            399999translated text
          </Text>
        </View>
        <View
          style={
            {
              "backgroundColor": "white",
              "height": 1,
              "marginBottom": 10,
              "marginTop": 15,
              "width": "100%",
            }
          }
        />
        <View
          style={
            {
              "display": "flex",
              "flexDirection": "row",
              "justifyContent": "space-between",
              "marginTop": 10,
              "width": "100%",
            }
          }
        >
          <Text
            style={
              {
                "color": "rgb(0, 122, 255)",
              }
            }
          >
            1500100
          </Text>
          <Text
            style={
              {
                "color": "rgb(0, 122, 255)",
              }
            }
          >
            1900100
          </Text>
        </View>
        <View
          style={
            {
              "display": "flex",
              "flexDirection": "row",
              "justifyContent": "space-between",
              "width": "100%",
            }
          }
        >
          <View
            style={
              {
                "borderLeftColor": "rgb(0, 122, 255)",
                "borderLeftWidth": 1,
                "height": 10,
              }
            }
          />
          <View
            style={
              {
                "borderRightColor": "rgb(0, 122, 255)",
                "borderRightWidth": 1,
                "height": 10,
              }
            }
          />
        </View>
        <View
          style={
            {
              "borderBottomColor": "rgb(0, 122, 255)",
              "borderBottomWidth": 2,
              "display": "flex",
              "flexDirection": "row",
              "justifyContent": "flex-start",
              "marginBottom": 0,
              "width": "100%",
            }
          }
        >
          <View
            style={
              {
                "backgroundColor": "lightyellow",
                "borderLeftColor": "rgb(0, 122, 255)",
                "borderLeftWidth": 1,
                "borderRightColor": "lightyellow",
                "borderRightWidth": 1,
                "height": 10,
                "width": "24.99981249953125%",
              }
            }
          />
          <View
            style={
              {
                "backgroundColor": "orange",
                "borderRightColor": "orange",
                "borderRightWidth": 1,
                "height": 10,
                "width": "50.0001250003125%",
              }
            }
          />
          <View
            style={
              {
                "backgroundColor": "#333333",
                "borderRightColor": "rgb(0, 122, 255)",
                "borderRightWidth": 1,
                "height": 10,
                "width": "25.00006250015625%",
              }
            }
          />
        </View>
        <View
          style={
            {
              "alignItems": "center",
              "display": "flex",
              "flexDirection": "row",
              "justifyContent": "flex-start",
              "marginTop": 5,
              "width": "100%",
            }
          }
        >
          <Text
            style={
              {
                "color": "rgb(0, 122, 255)",
              }
            }
          >
            translated text
          </Text>
          <View
            style={
              {
                "backgroundColor": "lightyellow",
                "display": "flex",
                "flexDirection": "row",
                "height": 10,
                "justifyContent": "flex-start",
                "margin": 5,
                "width": 10,
              }
            }
          />
          <Text
            style={
              {
                "color": "rgb(28, 28, 30)",
              }
            }
          >
            99999translated text25.00%
          </Text>
        </View>
        <View
          style={
            {
              "alignItems": "center",
              "display": "flex",
              "flexDirection": "row",
              "justifyContent": "flex-start",
              "marginTop": 5,
              "width": "100%",
            }
          }
        >
          <Text
            style={
              {
                "color": "rgb(0, 122, 255)",
              }
            }
          >
            translated text
          </Text>
          <View
            style={
              {
                "backgroundColor": "orange",
                "display": "flex",
                "flexDirection": "row",
                "height": 10,
                "justifyContent": "flex-start",
                "margin": 5,
                "width": 10,
              }
            }
          />
          <Text
            style={
              {
                "color": "rgb(28, 28, 30)",
              }
            }
          >
            200000translated text50.00%
          </Text>
        </View>
        <View
          style={
            {
              "alignItems": "center",
              "display": "flex",
              "flexDirection": "row",
              "justifyContent": "flex-start",
              "marginTop": 5,
              "width": "100%",
            }
          }
        >
          <Text
            style={
              {
                "color": "rgb(0, 122, 255)",
              }
            }
          >
            translated text
          </Text>
          <View
            style={
              {
                "backgroundColor": "#333333",
                "display": "flex",
                "flexDirection": "row",
                "height": 10,
                "justifyContent": "flex-start",
                "margin": 5,
                "width": 10,
              }
            }
          />
          <Text
            style={
              {
                "color": "rgb(28, 28, 30)",
              }
            }
          >
            100000translated text25.00%
          </Text>
        </View>
        <View
          style={
            {
              "backgroundColor": "white",
              "height": 2,
              "marginTop": 15,
              "width": "100%",
            }
          }
        />
        <View
          style={
            {
              "display": "flex",
              "marginTop": 20,
            }
          }
        >
          <Text
            style={
              {
                "color": "rgb(28, 28, 30)",
                "opacity": 0.65,
              }
            }
          >
            translated text
          </Text>
          <View
            style={
              {
                "width": 10,
              }
            }
          />
          <Text
            style={
              {
                "color": "rgb(28, 28, 30)",
                "fontSize": 18,
                "fontWeight": "600",
                "opacity": 1,
              }
            }
          >
            translated text5translated text50
          </Text>
        </View>
        <View
          style={
            {
              "display": "flex",
              "marginTop": 20,
            }
          }
        >
          <Text
            style={
              {
                "color": "rgb(28, 28, 30)",
                "opacity": 0.65,
              }
            }
          >
            translated text
          </Text>
          <View
            style={
              {
                "width": 10,
              }
            }
          />
          <Text
            style={
              {
                "color": "rgb(28, 28, 30)",
                "fontSize": 18,
                "fontWeight": "600",
                "opacity": 1,
              }
            }
          >
            100
          </Text>
        </View>
        <View
          style={
            {
              "alignItems": "flex-end",
              "flexDirection": "row",
            }
          }
        >
          <View
            style={
              {
                "display": "flex",
                "marginTop": 20,
              }
            }
          >
            <Text
              style={
                {
                  "color": "rgb(28, 28, 30)",
                  "opacity": 0.65,
                }
              }
            >
              translated text
            </Text>
            <View
              style={
                {
                  "width": 10,
                }
              }
            />
            <Text
              style={
                {
                  "color": "rgb(28, 28, 30)",
                  "fontSize": 18,
                  "fontWeight": "600",
                  "opacity": 1,
                }
              }
            >
              122
            </Text>
          </View>
          <View
            style={
              {
                "margin": 10,
              }
            }
          >
            <RNSVGSvgView
              bbHeight="20"
              bbWidth="20"
              focusable={false}
              height={20}
              style={
                [
                  {
                    "backgroundColor": "transparent",
                    "borderWidth": 0,
                  },
                  {
                    "flex": 0,
                    "height": 20,
                    "width": 20,
                  },
                ]
              }
              width={20}
            >
              <RNSVGGroup
                fill={
                  {
                    "payload": 4278190080,
                    "type": 0,
                  }
                }
              >
                <RNSVGCircle
                  cx="10"
                  cy="10"
                  fill={null}
                  propList={
                    [
                      "fill",
                      "stroke",
                      "strokeWidth",
                    ]
                  }
                  r="9.5"
                  stroke={
                    {
                      "payload": 4294967295,
                      "type": 0,
                    }
                  }
                  strokeWidth="1"
                />
                <RNSVGCircle
                  cx="10"
                  cy="10"
                  fill={null}
                  matrix={
                    [
                      6.123233995736766e-17,
                      -1,
                      1,
                      6.123233995736766e-17,
                      0,
                      20,
                    ]
                  }
                  propList={
                    [
                      "fill",
                      "stroke",
                      "strokeWidth",
                      "strokeDasharray",
                      "strokeDashoffset",
                      "strokeLinecap",
                    ]
                  }
                  r="9.5"
                  stroke={
                    {
                      "payload": 4278221567,
                      "type": 0,
                    }
                  }
                  strokeDasharray={
                    [
                      "59.690260418206066",
                      "59.690260418206066",
                    ]
                  }
                  strokeDashoffset={47.752208334564855}
                  strokeLinecap={1}
                  strokeWidth="1"
                />
                <RNSVGText
                  content={null}
                  dx={[]}
                  dy={[]}
                  fill={
                    {
                      "payload": 4280032286,
                      "type": 0,
                    }
                  }
                  font={
                    {
                      "fontSize": "12",
                      "textAnchor": "middle",
                    }
                  }
                  propList={
                    [
                      "fill",
                    ]
                  }
                  rotate={[]}
                  x={
                    [
                      "10",
                    ]
                  }
                  y={
                    [
                      "15",
                    ]
                  }
                >
                  <RNSVGTSpan
                    content=""
                    dx={[]}
                    dy={[]}
                    fill={
                      {
                        "payload": 4278190080,
                        "type": 0,
                      }
                    }
                    font={{}}
                    rotate={[]}
                    x={[]}
                    y={[]}
                  />
                </RNSVGText>
              </RNSVGGroup>
            </RNSVGSvgView>
          </View>
        </View>
        <View
          style={
            {
              "backgroundColor": "rgb(0, 122, 255)",
              "height": 2,
              "marginTop": 10,
              "width": "100%",
            }
          }
        />
        <View
          style={
            {
              "display": "flex",
              "marginTop": 20,
            }
          }
        >
          <Text
            style={
              {
                "color": "rgb(28, 28, 30)",
                "opacity": 0.65,
              }
            }
          >
            translated text
          </Text>
          <View
            style={
              {
                "width": 10,
              }
            }
          />
          <Text
            style={
              {
                "color": "rgb(28, 28, 30)",
                "fontSize": 18,
                "fontWeight": "600",
                "opacity": 1,
              }
            }
          >
            translated text1800100translated text1900100
          </Text>
        </View>
      </View>
    </View>
  </RCTScrollView>
  <View
    style={
      {
        "alignItems": "center",
        "flexDirection": "row",
        "flexGrow": 1,
        "justifyContent": "center",
        "marginVertical": 5,
      }
    }
  >
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "borderColor": "rgb(0, 122, 255)",
          "borderRadius": 10,
          "borderWidth": 2,
          "justifyContent": "center",
          "maxWidth": "90%",
          "minHeight": 48,
          "minWidth": "30%",
          "opacity": 1,
          "padding": 0,
          "paddingLeft": 20,
          "paddingRight": 20,
        }
      }
    >
      <View
        style={
          {
            "alignItems": "center",
            "display": "flex",
            "flexDirection": "row",
            "justifyContent": "center",
            "margin": 0,
            "minHeight": 48,
            "minWidth": 48,
            "padding": 0,
          }
        }
      >
        <Text
          style={
            {
              "color": "rgb(0, 122, 255)",
              "fontSize": 16,
              "fontWeight": "bold",
              "textTransform": "uppercase",
            }
          }
        >
          translated text
        </Text>
      </View>
    </View>
  </View>
</RCTSafeAreaView>
`;
